### PR TITLE
Fix for #300

### PR DESCRIPTION
**Pull request by jimbxb**
Regarding the new test case, I believe in the future we should raise an error if there is a public proc with a parameter whose type is not public.

The test case has such, with `hidden`

### CODE REVIEW

File: src/Types.hs [hunk 1]
```diff
@@ -540,20 +544,24 @@ instance Show Typing where
        then " (with no errors)"
        else " with errors: " ++ show errs
 
+lookupTyped :: String -> OptPos -> TypeSpec -> Typed TypeSpec
```
Comment (pschachte) [inline src/Types.hs:547]: Could you add a bit of doc?  The name is a bit mysterious to me.

File: test-cases/final-dump/unexported_type_overload.wybe [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module sub {
+    pub def proc(x:char) { pass }
+
+    pub def proc(x:hidden) { pass }
```
Comment (pschachte) [inline test-cases/final-dump/unexported_type_overload.wybe:4]: I thought this PR was meant to give an error message in this case?

**Review by pschachte — CHANGES_REQUESTED** [review]

File: test-cases/final-dump/unexported_type_overload.wybe (hunk 2 above)
Comment (jimbxb) [reply test-cases/final-dump/unexported_type_overload.wybe:4]: Yeah, so all this PR does is remove a spurious error.

As there is a correct overload (the `char` one), all is okay with no error.

I can add a test case to show the error is reported if the overload fails

Comment (jimbxb) [reply test-cases/final-dump/unexported_type_overload.wybe:4]: Nevermind, I found a way. Its a hack and a half around the type system though..

**Review by pschachte — APPROVED** [review]

**Pull request merged**